cluster-provision/images/vm-image-builder/customize-image.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+set -exuo pipefail
+
+function cleanup() {
+  echo "[customize-image] cleanup"
+  rm -rf "$CLOUD_INIT_ISO"
+  virsh destroy "$DOMAIN_NAME" || true
+  virsh undefine "$DOMAIN_NAME" || true
+
+  if [ $? -ne 0 ]; then
+    rm -f "$NEW_IMAGE"
+  fi
+}
+
+IMAGE_PATH=$1
+NEW_IMAGE=$2
+CLOUD_CONFIG_PATH=$3
+
+export OS_VARIANT=${OS_VARIANT:-fedora31}
+
+readonly DOMAIN_NAME="provision-vm"
+readonly CLOUD_INIT_ISO="cloudinit.iso"
+
+if ! [ -f "${CLOUD_CONFIG_PATH}" ]; then
+  CLOUD_CONFIG_PATH="user-cloud-config"
+fi
+
+trap 'cleanup' EXIT SIGINT
+
+# Create cloud-init user data ISO
+cloud-localds $CLOUD_INIT_ISO $CLOUD_CONFIG_PATH
+
+echo "Customize image by booting a VM with
+ the image and cloud-init disk
+ press ctrl+] to exit"
```

Comment:
> press ctrl+] to exit

But you do not need this as you will not stay on the console, right?

Reply:
You can run this script only, and you didnt add `sudo shutdown` in cloud-config file at `runCmd`
you will stay connected to the console until you close it manually.
This is good if you want to check what is the state of the image on live VM after cloud init provision.